src/Microsoft.Extensions.FileProviders.Physical/PhysicalFileProvider.cs

```diff
@@ -247,24 +248,16 @@ public IDirectoryContents GetDirectoryContents(string subpath)
 
         public IChangeToken Watch(string filter)
         {
-            if (filter == null)
+            if (filter == null || HasInvalidFilterChars(filter))
             {
                 return NullChangeToken.Singleton;
             }
 
-            if (PathNavigatesAboveRoot(filter))
-            {
-                return NullChangeToken.Singleton;
-            }
+            // Relative paths starting with leading slashes are okay
+            filter = filter.TrimStart(_pathSeparators);
 
-            // Relative paths starting with a leading slash okay
-            if (filter.StartsWith("/", StringComparison.Ordinal))
-            {
-                filter = filter.Substring(1);
-            }
-
-            // Absolute paths not permitted.
-            if (Path.IsPathRooted(filter))
+            // Absolute paths and paths traversing above root not permitted.
+            if (Path.IsPathRooted(filter) || PathNavigatesAboveRoot(filter))
```

Comment:
Moved the check for navigation above root to after trimming leading slashes which is how we validate for files and directories. Not sure why this was done before trimming previously.